### PR TITLE
lightning: move lightning directory out of br

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #52182

Problem Summary:

### Check List

Tests 

- [x] Unit test: existing test should cover it
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/lightning/log/log.go [hunk 1]
```diff
@@ -94,6 +94,7 @@ func InitLogger(cfg *Config, _ string) error {
 			return NewFilterCore(
 				core,
 				"github.com/pingcap/tidb/br/",
+				"/lightning/",
```
Comment (D3Hunter) [inline pkg/lightning/log/log.go:97]: one drawback of moving out, we need more log filter item

**Comment by D3Hunter** [comment]
lightning ci pass https://do.pingcap.net/jenkins/job/pingcap/job/tidb/job/pull_lightning_integration_test/146/display/redirect

File: Makefile [hunk 2]
```diff
@@ -314,20 +314,20 @@ bench-daily:
 
 build_tools: build_br build_lightning build_lightning-ctl
 
-br_web:
-	@cd br/web && npm install && npm run build
+lightning_web:
```
Comment (lance6716) [inline Makefile:317]: also need to notice CD team about the makefile entry changes

**Review by lance6716 — APPROVED** [review]

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Comment by D3Hunter** [comment]
br ci pass https://do.pingcap.net/jenkins/job/pingcap/job/tidb/job/pull_br_integration_test/90/display/redirect

**Review by 3pointer — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (D3Hunter) [reply Makefile:317]: seems it's used by developers to generate some files, like `checkpoints.pb.go` and `res_vfsdata.go`

Comment (wuhuizuo) [reply Makefile:317]: @D3Hunter 
CD will call `build_tools` task, so keep the command and output same will ensure the CD work well.


Thank you @lance6716 !

**Comment by wuhuizuo** [comment]
@D3Hunter Can you also move the test files to the new folder, or you planed to do it in another PR?

**Comment by D3Hunter** [comment]
> @D3Hunter Can you also move the test files to the new folder, or you planed to do it in another PR?

in later pr

**Pull request merged**